### ISSUE TITLE
Port to clang 11 and clang12

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#182]
A reminder: thank to the SFT team, clang11 and clang12 with OMP are now installed.
https://sft.its.cern.ch/jira/browse/SPI-1875

I should just try them out (and remove the hacks for missing OMP in the previous installation)

**Comment by valassi** [comment]
Possibly fix issue #172 first

**Comment by valassi** [comment]
This is now fully done.

I have tested that the latest installs of clang 11.1 and clang12.0 on cvmfs work fine, and have openmp. I removed the hack to disable openmp in clang 11.0 (only applies to that cvmfs installation).

En passant I tested the vectorisation performance of clang 11.1 and clang 12.0:
- clang 11.1 is similar to clang 10.0: it is a bit faster than gcc9 with vectorization! peak of 5.1E6 against 4.9E6, more or less
- clang 12.0 is (significantly slower) than the other two
- the difference between AVX2 and 512y is very tiny in all cases (avx2 faster in clang10/11, 512y faster in clang12, but slower than the other clangs anyway): for the moment I keep avx2 as the default on clang, unlike gcc where I keep 512y

I have also tested whether clang12.0 allow the use of cxtype_ref (issue #172), i.e. if now operator[] on vector extensions returns a non-const float, but it does not... I keep the other implementation on clang without cxtype_ref

A few minor patches, comments and logs are in PR #187 

This is fully clarified and can be closed
